src/utils/generator.js

```diff
@@ -1017,8 +1022,12 @@ export class Generator {
     this._addEntry({
       args: [joinCondition],
       buildNode: function([cond], context, valuesToProcess) {
-        let block1 = generator1.empty() ? null : serializeBody(generator1, context, valuesToProcess);
-        let block2 = generator2.empty() ? null : serializeBody(generator2, context, valuesToProcess);
+        let block1 = generator1.empty()
+          ? null
+          : Path.withCondition(joinCondition, () => serializeBody(generator1, context, valuesToProcess));
```

Comment:
The realm pathCondition isn't scoped right regarding the serializer. Conditions pushed while processing global could would linger around while the serializer is processing the generators of an optimized functions.